### PR TITLE
feat: run prerequest/postresponse actions when fetching gql schema

### PR DESCRIPTION
**Pull request by cwilper**
This PR addresses the following feature request:
* https://github.com/usebruno/bruno/issues/704

It automatically runs the usual pre-request and post-response code (vars and scripts) when fetching the GQL schema. It also tries to consolidate some of the relevant code to make re-use more manageable.

### Contribution Checklist:

- [x] **The pull request only addresses one issue or adds one feature.**
- [x] **The pull request does not introduce any breaking changes**
- [x] **I have read the [contribution guidelines](https://github.com/usebruno/bruno/blob/main/contributing.md).**
- [x] **Create an issue and link to the pull request.**

### CODE REVIEW

**Comment by helloanoop** [comment]
Thanks for taking care of this @cwilper !

> It also tries to consolidate some of the relevant code to make re-use more manageable.

Makes sense. This should help us move this logic seamlessly to a separate package (post v1 release) to be shared across electron and cli.

File: packages/bruno-electron/src/ipc/network/index.js [hunk 1]
```diff
@@ -502,8 +540,25 @@ const registerNetworkIpc = (mainWindow) => {
         });
       }
 
-      const processEnvVars = getProcessEnvVars(collection.uid);
-      interpolateVars(preparedRequest, envVars, collection.collectionVariables, processEnvVars);
```
Comment (mheidinger) [inline packages/bruno-electron/src/ipc/network/index.js:506]: This unfortunately breaks interpolation for the introspection query.
The `runPreRequest` and with that the `interpolateVars` is executed on `request`, but actually executed is `preparedRequest` from https://github.com/usebruno/bruno/pull/705/files#diff-eaeb1da31271859ebecb449a3291ab31d513dbee71aff0df896001827f94cc90R533 in https://github.com/usebruno/bruno/pull/705/files#diff-eaeb1da31271859ebecb449a3291ab31d513dbee71aff0df896001827f94cc90R570.

Not sure how to properly fix this best 🤔

File: packages/bruno-electron/src/ipc/network/index.js [hunk 2]
```diff
@@ -514,6 +569,19 @@ const registerNetworkIpc = (mainWindow) => {
       );
       const response = await axiosInstance(preparedRequest);
 
+      await runPostResponse(
```
Comment (mheidinger) [inline packages/bruno-electron/src/ipc/network/index.js:572]: On first thought I am not sure if running post request stuff makes sense.
At least my expectation is that post request scripts are written to run on the actual response of the request (i.e. extract data from the response, run assertions on the response). If now suddenly the response is the introspection data, I'd expect that those scripts and assertions will definitively fail or result in unexpected behaviour.

File: packages/bruno-electron/src/ipc/network/index.js (hunk 1 above)
Comment (mheidinger) [reply packages/bruno-electron/src/ipc/network/index.js:506]: Tagging @helloanoop directly, as this should be fixed before the 1.0 release

Comment (helloanoop) [reply packages/bruno-electron/src/ipc/network/index.js:506]: @mheidinger 

I will spend some time next week, This file has gotten really hard to maintain. We have `prepareRequest` and `configureRequest`. Then we have configuring proxy, and the entire request lifecycle along with nuances around running single request, running a folder (collection run), and fetching gql schema.

Need to look at this more holistically. Ideally we should have just 1 separate file with a single`runRequest` method with a good configurability (ex: disable running assertions/tests etc)

For now, to solve this issue, can you update the `fetch-gql-schema` [ipc handler](https://github.com/usebruno/bruno/blob/main/packages/bruno-electron/src/ipc/network/index.js#L551-L626) and duplicate the code inside it needed to run request lifecycle stuff (pre request vars and script, post response vars and script)

@cwilper @mheidinger @grubersjoe If anyone of you have bandwidth, please pick this up.

Its ok to duplicate code.
We'll get to the ideal DRY version post release 1.0.0

Comment (mheidinger) [reply packages/bruno-electron/src/ipc/network/index.js:506]: Created https://github.com/usebruno/bruno/pull/809 to just add back in that one `interpolateVars` call

**Pull request merged**